### PR TITLE
Fix #353 Add dev mode in running the server and update installation guide

### LINKED ISSUE #353: Debug is True on deploying app to production to heroku
**I'm submitting a ...**
- [x] bug report
- [ ] feature request

**Current behavior:**

**Expected behavior:**

Best practice is debug should be False on deploying app to production to heroku.

**Steps to reproduce:**

I am taking this issue to work on.
Maintainers do let me know if `Debug=True` is intended while deploying or not.

### PR DESCRIPTION
**Pull request by raju249**
Fixes #353 

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `master` branch.
- [x] I have added necessary documentation (if appropriate)

#### Changes proposed in this pull request:

- Default mode for debug is set to `False`.
- Add `--dev` command line flag while running in development mode

### CODE REVIEW

File: app/server.py [hunk 1]
```diff
@@ -88,10 +95,18 @@ def set_header(r):
...
-    app.run(
-        host='0.0.0.0',
-        port=int(
-            os.environ.get(
-                'PORT',
-                7001)),
-        debug=True)
+    if args.dev:
+        app.run(
+            host='0.0.0.0',
```
Comment (vaibhavsingh97) [inline app/server.py]: It's not human readable. Please fix it

File: app/server.py [hunk 2]
```diff
@@ -88,10 +95,18 @@ def set_header(r):
...
+            port=int(
+                os.environ.get(
+                    'PORT',
+                    7001)),
+            debug=True)
+    else:
+        app.run(
+            host='0.0.0.0',
+            port=int(
```
Comment (vaibhavsingh97) [inline app/server.py]: same

Comment (raju249) [reply app/server.py]: Alright!

File: app/server.py [hunk 3]
```diff
@@ -88,10 +95,14 @@ def set_header(r):
...
 if __name__ == '__main__':
-    app.run(
-        host='0.0.0.0',
-        port=int(
-            os.environ.get(
-                'PORT',
-                7001)),
-        debug=True)
+
+    if args.dev:
```
Comment (AnshulMalik) [inline app/server.py]: We can make it look good by not having an if block, maybe use oonditional statement above to define debug.

Comment (raju249) [reply app/server.py]: Sorry @AnshulMalik 
Didn't get you. Can you please elaborate more ?

Comment (AnshulMalik) [reply app/server.py]: Right now you are using if and else block, but both share the same code, what we can do here is, 

```python
app.run(
        host='0.0.0.0',
        port=int(
            os.environ.get(
                'PORT',
                7001)),
        debug=args.dev)
```

Comment (raju249) [reply app/server.py]: Awesome @AnshulMalik 
Got it. Missed it though :sweat_smile:

**Comment by raju249** [comment]
@vaibhavsingh97 
@AnshulMalik 
I have integrated the suggestions. Kindly review the PR.

**Pull request merged**